pkg/apis/serving/v1alpha1/inference_graph_validation.go

```diff
@@ -0,0 +1,120 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	//InvalideGraphNameFormatError defines the error message for invalid inference graph name
+	InvalidGraphNameFormatError = "The InferenceGraph \"%s\" is invalid: a InferenceGraph name must consist of lower case alphanumeric characters or '-', and must start with alphabetical character. (e.g. \"my-name\" or \"abc-123\", regex used for validation is '%s')"
+)
+
+// regular expressions for validation of isvc name
+const (
+	GraphNameFmt string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+)
+
+var (
+	// logger for the validation webhook.
+	validatorLogger = logf.Log.WithName("inferencegraph-v1alpha1-validation-webhook")
+	//GraphRegexp regular expressions for validation of graph name
+	GraphRegexp = regexp.MustCompile("^" + GraphNameFmt + "$")
+)
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-inferencegraph,mutating=false,failurePolicy=fail,groups=serving.kserve.io,resources=pods,versions=v1alpha1,name=inferencegraph.kserve-webhook-server.validator
+
+var _ webhook.Validator = &InferenceGraph{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (ig *InferenceGraph) ValidateCreate() error {
+	validatorLogger.Info("validate create", "name", ig.Name)
+
+	if err := validateInferenceGraphName(ig); err != nil {
+		return err
+	}
+
+	if err := validateInferenceGraphRouterType(ig); err != nil {
+		return err
+	}
+
+	if err := validateInferenceGraphSplitterWeight(ig); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (ig *InferenceGraph) ValidateUpdate(old runtime.Object) error {
+	validatorLogger.Info("validate update", "name:", ig.Name)
+
+	return ig.ValidateCreate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (ig *InferenceGraph) ValidateDelete() error {
+	validatorLogger.Info("validate delete", "name", ig.Name)
+	return nil
+}
+
+// Validation of inference graph name
+func validateInferenceGraphName(ig *InferenceGraph) error {
+	if !GraphRegexp.MatchString(ig.Name) {
+		return fmt.Errorf(InvalidGraphNameFormatError, ig.Name, GraphNameFmt)
+	}
+	return nil
+}
+
+//Validation of inference graph router type
+func validateInferenceGraphRouterType(ig *InferenceGraph) error {
+	nodes := ig.Spec.Nodes
+	for name, node := range nodes {
+		if node.RouterType != Single && node.RouterType != Splitter &&
```

Comment:
I think we can use kubebuilder enum annotation to validate this

Reply:
Yes, it is better use kubebuilder.